### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by wcmitchell**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-9182

## Description of Intent of Change(s)
There was a request to clean up our documentation and specifically note
that resourceDefinitions should not be used by non-cost-management apps.
Did some cleanup here, but I susupect we'll need to do more in the near
future for real clarity of how RBAC works.

## Local Testing
All docs, no code

## Checklist
- [x] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [ ] are theses changes covered by unit tests?

- [x] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

File: docs/source/specs/openapi.json [hunk 1]
```diff
@@ -1444,275 +1440,7 @@
         }
       }
     },
-    "/policies/": {
```
Comment (coderbydesign) [inline docs/source/specs/openapi.json:1447]: I wonder if we need to be concerned with any clients already build against this doc _with_ policies? Probably worth at least an email on official deprecation.

File: docs/source/management/role.rst [hunk 2]
```diff
@@ -23,12 +23,16 @@ Operation defines the application logic action, for example::
 - write
 - order
 
+Note that in any of the above stanzas, a '*' is taked to mean 'all'.
```
Comment (coderbydesign) [inline docs/source/management/role.rst:26]: Will this format weird? I see it's italicized. I haven't generated the docs yet locally to check for sure.

File: docs/source/management/role.rst [hunk 3]
```diff
@@ -23,12 +23,16 @@ Operation defines the application logic action, for example::
...
+
 
 Resource Definitions
 ********************
-If an empty array is supplied for resource definitions then this is implied to mean all access. However, resource access can be limited by specifing an attribute filter.
+Resource Definitions are a somewhat trickier aspect of our implementation of RBAC. In general, ALL roles should be created with a resourceDefinitions stanza of '[]'. This is taken to mean "no additional filtering" and will generally result in the expected behavior.
```
Comment (astrozzc) [inline docs/source/management/role.rst:31]: Nice! Overall LGTM! But seems the description that resource definitions only apply to cost-management is missing here

Comment (wcmitchell) [reply docs/source/management/role.rst:31]: Ah, I wasn't sure whether to put that in specifically, I'll get it added!

File: docs/source/management/role.rst (hunk 2 above)
Comment (wcmitchell) [reply docs/source/management/role.rst:26]: That's a good thought, my RST preview didn't show anything, but will test locally just to be sure.

File: docs/source/specs/openapi.json (hunk 1 above)
Comment (wcmitchell) [reply docs/source/specs/openapi.json:1447]: Good thought, will handle that Monday.

**Review by astrozzc — APPROVED** [review]
LGTM

**Review by coderbydesign — APPROVED** [review]

**Pull request merged**